submissions/slack-status/src/background.js

```diff
@@ -0,0 +1,102 @@
+import { getSlackAuth } from "./slack.js";
+
+async function main() {
+  let auth;
+  try {
+    auth = await getSlackAuth();
+  } catch (e) {
+    console.error(e);
+    await chrome.tabs.create({ url: chrome.runtime.getURL("src/settings.html") });
+    return;
+  }
+
+  const { enabled, authorized } = await chrome.storage.local.get({ enabled: true, authorized: false });
+  if (!authorized) {
+    await chrome.tabs.create({ url: chrome.runtime.getURL("src/authorize.html") });
+  }
+
+  let lastTabId = -1;
+  while (true) {
```

Comment:
When I have the extension running my laptop gets warm. I suspect this while loop is the culprit

Reply:
I actually used the chrome tabchange events before, but it unfortunately wouldn’t detect a lot of tab changes (pretty sure you ran into that a while ago) so this is necessary. I can increase the sleep time though.

Reply:
Updated timeout!

Reply:
I worked on a similar extension during stream some time ago. You could just use a `setInterval` for this. https://github.com/JosiasAurel/website-presence-slack/blob/4f38591c2c82d0952b3c6478fd0fbf8bcc03e743/update-slack-status.js#L32